src/functions/updateDmsTaskWithDates/framework/dms/table-mapping.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * Handles generating DMS table mappings from a much simpler logical input.
+ */
+import { getLogger } from '../util';
+import { DateTime } from 'luxon';
+
+const logger = getLogger('table-mapping', 'debug');
+
+export interface OrCondition {
+  operator: string;
+  value?: string;
+  start?: string;
+  end?: string;
+}
+
+export interface AndFilter {
+  column: string;
+  orConditions: OrCondition[];
+}
+
+export interface Table {
+  sourceName: string;
+  andFilters?: AndFilter[];
+  removeColumns?: string[];
+}
+
+export interface Options {
+  sourceSchema: string;
+  destSchema: string;
+  tables: Table[];
+}
+
+export function generateTableMapping(options: Options): any {
```

Comment:
more safety around the return type perhaps?

Reply:
provided simple type safety